flight/PiOS/STM32F4xx/pios_reset.c

```diff
@@ -0,0 +1,50 @@
+/**
+ ******************************************************************************
+ * @addtogroup PIOS PIOS Core hardware abstraction layer
+ * @{
+ * @addtogroup PIOS_RESET Reset functions
+ * @brief Hardware functions to deal with the reset register
+ * @{
+ *
+ * @file       pios_reset.c
+ * @author     Kenn Sebesta, Copyright (C) 2015.
+ * @brief      Reset information
+ *
+ ******************************************************************************/
+
+
+/* Project Includes */
+#include "pios.h"
+
+#include "pios_reset.h"
+
+void PIOS_RESET_Clear()
+{
+    RCC_ClearFlag();
+}
+
+int16_t PIOS_RESET_GetResetReason (void)
+{
+	uint8_t reboot_reason = PIOS_RESET_FLAG_UNDEFINED;
+	
+	// ****************************************************** //
+	// The order of these checks is important. DO NOT CHANGE. //
+	// ****************************************************** //
+	if(RCC_GetFlagStatus(RCC_FLAG_PORRST) == SET) {         // Check #1
```

Comment:
How would that work in the ```switch()..case```? ```RCC_GetFlagStatus``` has to be called with an argument so I will have to iterate over each possible reset reason.

I could write my own function to grab the RCC flag directly, but that's code duplication and targets the specific processor instead of using the CMSIS abstraction, right?